pydatastructs/miscellaneous_data_structures/stack.py

```diff
@@ -64,16 +64,16 @@ def pop(self, *args, **kwargs):
 
     @property
     def is_empty(self):
-        return None
+        raise NotImplementedError(
+              "This is an abstract method.")
 
     @property
     def peek(self):
-        return None
+        raise NotImplementedError(
+              "This is an abstract method.")
 
 class ArrayStack(Stack):
 
-    __slots__ = ['items']
```

Comment:
Why?

Reply:
It wasn't doing anything since the initialization is already being done in the __new__() itself. All tests are passing without it.

Reply:
See, http://book.pythontips.com/en/latest/__slots__magic.html
Please restore this.

Reply:
Oh I didn't realize it optimized memory usage also, I am sorry I will solve it right away.